core/os/thread.h

```diff
@@ -102,6 +106,8 @@ class Thread {
 	///< waits until thread is finished, and deallocates it.
 	void wait_to_finish();
 
+	static void attach_external_thread();
```

Comment:
How should this be used?
Should this be exposed to scripting and users are expected to call this when they start threads?
At least in C# when using tasks there is no way of doing this without either hooking various runtime internals or just sprinkling calls to this everywhere, in which case my caller_id suggestion would be way more efficient.

Reply:
I after thinking about it, your above suggestion is likely far safer. I think its common in situations, as an example, that the audio thread is created by the audio driver (not by the engine) and things like that, so will do that change.